tensorflow/compiler/xla/service/gpu/gpu_conv_runner.cc

```diff
@@ -261,8 +270,13 @@ StatusOr<GpuConvParams> GetGpuConvParams(
   const Shape* filter_shape;
   const Shape* output_shape;
 
-  params.algorithm = se::dnn::AlgorithmConfig(se::dnn::AlgorithmDesc(
-      backend_config.algorithm(), backend_config.tensor_ops_enabled()));
+  // The third field is scratch size stored from conv_algorithm_picker
+  // The operand is added to the shape field of the conv instruction
+  // in GpuConvAlgorithmPicker::RunOnInstruction() call.
+  params.algorithm = se::dnn::AlgorithmConfig(
+      se::dnn::AlgorithmDesc(backend_config.algorithm(),
+                             backend_config.tensor_ops_enabled()),
+      conv->shape().tuple_shapes(1).dimensions(0));
```

Comment:
Why the extra argument?

Reply:
This additional field is the scratch_size. When an algorithm is specified in `ROCm`, `ThenConvolveWithAlgorithm()` will expect the scratch_size to be set, [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/stream_executor/rocm/rocm_dnn.cc#L2737). Without this field, ROCm stream executor won't be able to know how much scratch size to allocate.